fe/src/main/java/org/apache/doris/qe/QueryDetail.java

```diff
@@ -0,0 +1,114 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.qe;
+
+public class QueryDetail {
+    public enum QueryMemState {
+        RUNNING,
+        FINISHED,
+        FAILED,
+        CANCELLED 
+    };
+
+    // When query received, FE will construct a QueryDtail  
+    // object. This object will set queryId, startTime, sql
+    // fields. As well state is be set as RUNNING. 
+    // After query finished, endTime and latency will
+    // be set and state will be updated to be FINISHED/FAILED/CANCELLED
+    // according to the query execution results.
+    // So, one query will be inserted into as a item and 
+    // be udpated upon finished. To indicate the two event,
+    // a extra field named eventTime is added.
+    private long eventTime;
```

Comment:
I still don't understand what's the meaning of eventTime?  and why do we need it?

Reply:
Every query will have two event.
When starts a query, it will set QueryMemState as RUNNING.
When finished, it will set QueryMemState as FINISHED/FAILED/CANCELLED according to query execution situation.
QueryMonitor will collect QueryMemState starts a event time to handle this situation.